app/src/main/java/protect/card_locker/importexport/CatimaImporter.java

```diff
@@ -73,16 +73,12 @@ public void importData(Context context, SQLiteDatabase database, InputStream inp
     public void importCSV(Context context, SQLiteDatabase database, InputStream input) throws IOException, FormatException, InterruptedException {
         BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(input, StandardCharsets.UTF_8));
 
-        bufferedReader.mark(100);
-
-        Integer version = 1;
-
-        try {
-            version = Integer.parseInt(bufferedReader.readLine());
-        } catch (NumberFormatException _e) {
-            // Assume version 1
-        }
-
+        bufferedReader.mark(10);
+        int codePoint = bufferedReader.read();
+        // default to version 1
+        // keep in mind this breaks with more digits, but is probably better than BufferedReader.readLine
```

Comment:
How many digits is "more digits"?

Reply:
More than one digit. If you go into version 10 or above, you'll need to add code to parse the other digit. But imo it's better than accidentally going over the mark buffer by using readline.

Reply:
Given version should always be on its own line, wouldn't it make sense to read until the newline character (and throw an exception if we reach the mark before that) so this couldn't ever break? I find it quite scary to think of backup import silently breaking some time far far into the future by trying to use the wrong version because the format changed too often (even though it's unlikely to happen)

Reply:
Realistically you'll have to edit the same class first. But sure I'll add that in.